src/server.rs

```diff
@@ -519,3 +724,149 @@ impl Drop for ServerManager {
         unsafe { sys::SteamGameServer_Shutdown() }
     }
 }
+
+/// Called when a client has been approved to connect to this game server
+#[derive(Clone, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct GSClientApprove {
+    /// The steam ID of the user requesting a p2p session
+    pub user: SteamId,
+    /// Owner of the game, may be different from user when Family Sharing is being used
+    pub owner: SteamId,
+}
+
+unsafe impl Callback for GSClientApprove {
+    const ID: i32 = 201;
```

Comment:
Instead of using the magic constant here, let's use the corresponding `steamworks_sys` constant instead. Same with the other callbacks below.

Reply:
Ah I remember why I didn't use the constants now. They're super ugly in the binding, with the `i32` value being wrapped in an enum. I can do this for now, but ideally the binding code would be cleaner.

```
    const ID: i32 = steamworks_sys::GSClientApprove_t_k_iCallback as i32;
```